DSpace/dspace-angular
# Edit item page permission checks

## References
* Fixes #1073 
* REST API PR: https://github.com/DSpace/DSpace/pull/3240

## Description
This PR adds more fine-grained permission checks to the edit item page.

## Instructions for Reviewers
Changes made:
* Added new features to FeatureID enum: canManageVersions, canManageBitstreams, canManageRelationships, canManageMappings, canManagePolicies, canMakePrivate, canMove
* Created guards for every tab on the item edit page and every action on the status tab. Some of these guards are more advanced (e.g. the guard for the status tab requires at least one of the actions found on the page to be authorized)
* Unauthorized actions on the status page will contain a disabled button with a tooltip when hovering over it mentioning they're not authorized to perform the action

Mocks:
Since we're [waiting on the REST PR](https://github.com/DSpace/DSpace/issues/3201) to add some of the feature endpoints, I've added mocks for these endpoints on certain items. Rest api containing these items: https://api7.dspace.org/server/#/server/api
* Authorization for `canManageBitstreams` is mocked for item 96715576-3748-4761-ad45-001646632963
* Authorization for `canManageRelationships` is mocked for item 047556d1-3d01-4c53-bc68-0cee7ad7ed4e
* Authorizations for `canManageVersions` and `canManageMappings` are mocked for item e98b0f27-5c19-49a0-960d-eb6ad5287067

How to test:
* Visit the edit item page for the following items:
** 96715576-3748-4761-ad45-001646632963: The bitstreams tab should be available. Relationships, versions and collection mapper should be inaccessible. The collection mapper action on the status page should also be disabled and show a tooltip when hovered over.
** 047556d1-3d01-4c53-bc68-0cee7ad7ed4e: The relationships tab should be available. Bitstreams, versions and collection mapper should be inaccessible. The collection mapper action on the status page should also be disabled and show a tooltip when hovered over.
** e98b0f27-5c19-49a0-960d-eb6ad5287067: The versions and collection mapper tabs should be available. Bitstreams and relationships should be inaccessible. The collection mapper action on the status page should be active and link to the collection mapper tab.
** Visiting any other item's edit page should have all the above actions disabled

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
    * Include tests for different user types (if behavior differs), including: (1) Anonymous user, (2) Logged in user (non-admin), and (3) Administrator.
    * Include tests for error scenarios, e.g. when errors/warnings should appear (or buttons should be disabled).
    * For bug fixes, include a test that reproduces the bug and proves it is fixed. For clarity, it may be useful to provide the test in a separate commit from the bug fix.
